rpc/handle_get_block_template.go

```diff
@@ -770,3 +770,21 @@ func decodeLongPollID(longPollID string) ([]*daghash.Hash, int64, error) {
 
 	return parentHashes, lastGenerated, nil
 }
+
+// isSynced checks if the node is synced enough based upon its worldview.
+// This is used to determine if the node can support mining and requesting newly-mined blocks.
+// To do that, first it checks if the selected tip timestamp is not older than maxTipAge. If that's the case, it means
+// the node is synced since blocks' timestamps are not allowed to deviate too much into the future.
+// If that's not the case it checks the rate it added new blocks to the DAG recently. If it's faster than
+// blockRate * maxSyncRateDeviation it means the node is not synced, since when the node is synced it shouldn't add
+// blocks to the DAG faster than the block rate.
+func isSynced(s *Server) bool {
```

Comment:
This is very business-logicy, shouldn't be in the RPC package.
Please either move to blockdag (or wherever relavant), or create a ticket